src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -234,17 +229,40 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
                         result.append(".git");
                         break;
                     case SSH:
-                        result.append("ssh://git@");
-                        result.append(url.getHost());
-                        if (protocolPortOverride  != null && protocolPortOverride > 0) {
-                            result.append(':');
-                            result.append(protocolPortOverride);
+                        URI cloneLinkUri = null;
+                        if (cloneLink != null) {
+                            try {
+                                cloneLinkUri = new URI(cloneLink);
+                            } catch (@SuppressWarnings("unused") URISyntaxException ignored) {
+                                // fall through
+                            }
+                        }
+
+                        if (cloneLinkUri != null) {
+                            if (cloneLinkUri.getScheme() != null) {
+                                result.append(cloneLinkUri.getScheme()).append("://");
+                            } else {
+                                result.append("ssh://");
+                            }
+                            if (cloneLinkUri.getRawAuthority() != null) {
+                                result.append(cloneLinkUri.getRawAuthority());
+                            } else {
```

Comment:
This is the point where I would stop using StringBuilder and use Handy URI template